### PR TITLE
adds passenger seats to space pods

### LINKED ISSUE #14657: Feature request: Two seater pod
Pretty self explanatory from the title. Why go fuck off into space by yourself when you can go with a friend!

### PR DESCRIPTION
**Pull request by MadmanMartian**
**IT'S DONE AS FAR AS I CAN SEE**

Was looking through the WorkInProgress files, and found that spacepods came under this, so I'm going to be poking at them for a bit, assuming I have the permission of pomf.

First up is adding passenger seating to the space pod. It seems odd that something 64x64 would not have room for more than 1 person, especially considering when you can just as easily stack an infinite number of people into a 3x1 escape pod.

Also changed the enter/exit pod verb to the spacepod tab that's used when firing weapons, rather than it being in the objects tab.

If you've got any suggestions for other quality of life changes for spacepods, leave them in the comments section and I'll see what I can do.

**Requesting sprites**
- [x] A taxi-like space pod. Basically a recolor of the current space pod with New York taxi colouring. Primarily yellow with black and white checkerboard. - Thanks @Kurfursten !
- [x] Different pod armor piece for the taxi space pod, so it's acquirable in some way.
- [x] Action button sprites for - Thanks Bilgecrank in the discord!
 - pod weapon fire
 - toggling the passenger allowance
 - toggling passenger weaponry
 - passengers taking control of the pod because the pilot left

**Things to do**
- [x] Either take the trigger away from passengers entirely, or make it something the pilot can toggle and control (Gunners soon)
- [x] Add a way for pilots to disallow people from trying to be passengers, maybe even an ejector seat.
- [x] Add a tie in so that, if the pilot gets out, a passenger can opt in to take control instead.

closes #14657

:cl:
* rscadd: Adds passenger seats to spacepods
* rscadd: Adds action buttons for spacepods. Sprites courtesy of Bilgecrank
* rscadd: Adds alternate armor for spacepod that makes it into a taxi pod. More passenger seats, but no weapons. Sprites courtesy of Kurfursten

### CODE REVIEW

**Comment by Kurfursten** [comment]
"Fat checker" version
untitled

Alternate "skinny checker" version
untitled


I'm not a real spiter but what do you think?

**Comment by Exxion** [comment]
Pomf hasn't touched spacepods in...
Well before I started coding here

Please, yes, for the love of god, do whatever the fuck you want to spacepods
Nothing you could do could make them worse than they already are

**Comment by MadmanMartian** [comment]
@Kurfursten that seems a bit too bright and golden, new york taxis are a bit more towards the red-orange side. At least from what I can see, but I'm a tad toneblind so there's that

classic taxi

modern taxi

5th element taxi

**Comment by Kurfursten** [comment]
untitled

The original colors were actually taken from a photo of a Taxi

Here's a darker tone though

**Comment by MadmanMartian** [comment]
Still looks waaay too much like a pimpmobile. Maybe it's the white highlights, giving it too much of a golden look to it

**Comment by Dacendeth** [comment]
Would it be possible to increase your view distance while riding in a pod? It doesnt have to be anything drastic, and it could be default or an optional attachment you have to build/research. Also it be nice if there were some sort of lights on the pod so you can see walls near it.

**Comment by Intigracy** [comment]
That second one looks fine.

Also taxis obviously should hold 2/3 passengers instead of 1.

**Comment by Exxion** [comment]
The antialiasing or whatever on the word "TAXI" on it looks awful though

**Comment by Kurfursten** [comment]
Grey highlights + 2-color shading on TAXI

untitled

**Review by PJB3005 — CHANGES_REQUESTED** [review]
Occupants and passengers should be indistinguishable, except for the first passenger being pilot. No duplicating code to handle both occupant and passenger.

**Comment by MadmanMartian** [comment]
just found a better way of setting verbs, hang on.

@PJB3005 It's gonna use the alternate proc for assigning the pilot sadly

**Comment by PJB3005** [comment]
> @PJB3005 It's gonna use the alternate proc for assigning the pilot sadly

No offense but that just sounds like "I don't want to refactor"

**Comment by MadmanMartian** [comment]
I'm refactoring to remove the copypaste code in the two entry procs, but I need some way to have a distinct difference between pilots and passengers so I can have things affect each of them differently. 

Would you prefer I just have it so that passengers[1] be considered the pilot? Seems like it'd get a bit odd

**Comment by Exxion** [comment]
Yeah that would be a bad idea
You could have the pilot also be one of the passengers, though. They enter the pod as a passenger then are designated the pilot.
Similar system could be used for the gunner.

**Comment by Kurfursten** [comment]
You could use an associative list to define the role of each person inside, which could be useful in the case of setting up pods with different jobs. e.g.: pilot, gunner, passenger, navigator.

File: code/WorkInProgress/pomf/spacepods/spacepods.dm [hunk 1]
```diff
@@ -326,24 +326,47 @@
...
 	if(occupant)
 		inertia_dir = 0 // engage reverse thruster and power down pod
 		occupant.forceMove(exit_loc)
 		occupant = null
 		to_chat(usr, "<span class='notice'>You climb out of the pod.</span>")
+		if(passengers.len)
+			for(var/mob/living/L in passengers.len)
+				if(!occupant)
+
+					var/answer = alert(L,"The pilot has evacuated \the [src]. Do you wish to take over?",,"Yes","No")
```
Comment (MadmanMartian) [inline code/WorkInProgress/pomf/spacepods/spacepods.dm]: Requesting a second opinion on this. I can't test it by myself

**Comment by MadmanMartian** [comment]
Alright, that's all the verbs done as far as I'm aware. 

Time to see what I can do about the construction process to allow for alternate pod designs

EDIT: Which monster made construction datums

FURTHER EDIT: Seems mecha just copypastes each other anyway. The firefighter ripley and the regular ripley are copypastes of one another largely, so I assume that is acceptable?

File: code/WorkInProgress/pomf/spacepods/spacepods.dm (hunk 1 above)
Comment (Kurfursten) [reply code/WorkInProgress/pomf/spacepods/spacepods.dm]: You can!

Aghost out of your body once inside, then use the secrets panel to spawn yourself as a test dummy and load in as the second passenger. This trick is useful for all kinds of two-person testing you'll do over time.

Comment (MadmanMartian) [reply code/WorkInProgress/pomf/spacepods/spacepods.dm]: Problem is I need to be both the pilot and the passenger at the same time, as to test this I have to eject as the pilot, then see if the prompt pops up for the passenger

Comment (PJB3005) [reply code/WorkInProgress/pomf/spacepods/spacepods.dm]: Alert menus are pretty terrible, don't do this.

A message and verb or HUD button somewhere to take over is better.

**Comment by Shadowmech88** [comment]
>The firefighter ripley and the regular ripley are copypastes of one another largely, so I assume that is acceptable?

Sometimes code is the way it is because it's old and no one's fixed it, not because the way it is is acceptable.

File: code/WorkInProgress/pomf/spacepods/spacepods.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/WorkInProgress/pomf/spacepods/spacepods.dm]: How abouts do I add verbs only under certain conditions? I tried occupant.verbs += (verb), but that only seems to work for verbs defined under the subtype of the occupant, and not the spacepod.

Verbs seem like quite a mess, hence I want to UI it up a bit

Comment (PJB3005) [reply code/WorkInProgress/pomf/spacepods/spacepods.dm]: Action buttons.

Comment (Exxion) [reply code/WorkInProgress/pomf/spacepods/spacepods.dm]: Yes, use action buttons
But if you want an example of how to conditionally add verbs, check the spacepod equipment file
I don't remember how it's done but I know it's done in there

Comment (MadmanMartian) [reply code/WorkInProgress/pomf/spacepods/spacepods.dm]: new verb(), nasty stuff

File: code/WorkInProgress/pomf/spacepods/spacepods.dm [hunk 2]
```diff
@@ -514,5 +552,110 @@
...
+	to_chat(src.occupant, "<span class='notice'>Now [passengers_allowed?"allowing passengers":"disallowing passengers, and ejecting any current passengers"].</span>")
+	if(!passengers_allowed && passengers.len)
+		for(var/mob/living/L in passengers)
+			to_chat(L, "<span class='warning'>Ejection sequence activated: Ejecting in 3 seconds</span>")
+			spawn(30)
+				if(passengers.Find(L) && L.loc == src)
+					playsound(src, 'sound/weapons/rocket.ogg', 50, 1)
+					var/turf/T = get_turf(src)
+					var/turf/target_turf
+					L.forceMove(T)
```
Comment (MadmanMartian) [inline code/WorkInProgress/pomf/spacepods/spacepods.dm]: Feck, missed one. Gotta use move_passenger_outside for button handling

**Comment by MadmanMartian** [comment]
- Adds lists for action button handling for space pods
- Attempts to handle the quantum fuckload of references the pod holds when qdel'd, it just didn't seem right for all of those to be there
- Adds the taxi pod, thank you @Kurfursten !

**Comment by ghost** [comment]
A lot of space explorers wanted a cargo hold of some sort.

**Comment by MadmanMartian** [comment]
That was what I wanted to do next.

I wanted to just do bits at a time, now I've ended up rewriting half the damn thing

**Comment by MadmanMartian** [comment]
The absolute madman

**Pull request merged**